src/patterns.py

```diff
@@ -13,7 +13,10 @@ def pattern1(self, n):
         * * *
         * * * *
         """
-        raise Exception("Not Implemented!")
+        for x in range(n):
```

Comment:
you have now hard-coded the printing to use the '*' character. Try to introduce a named parameter in the method pattern1, so that it can be optionally invoked with some other character, and you can use that character to print the pattern.